src/core/control/tools/EllipseHandler.cpp

```diff
@@ -80,9 +79,7 @@ auto EllipseHandler::createShape(bool isAltDown, bool isShiftDown, bool isContro
 
     shape.emplace_back(center_x + radiusX, center_y);
     for (unsigned int j = 1U; j < nbPtsPerQuadrant; j++) {
-        const double tgtAngle = stepAngle * j;
-        // This formula dispatches the points, so that points are denser where the ellipse has higher curvature.
-        const double centerAngle = 0.5 * (std::atan(excentricity * std::tan(tgtAngle)) + tgtAngle);
+        const double centerAngle = stepAngle * j;
```

Comment:
We can also resolve this bug by using `atan2` instead of removing @bhennion's code.
```suggestion
        const double tgtAngle = stepAngle * j;
        // This formula dispatches the points, so that points are denser where the ellipse has higher curvature.
        const double centerAngle = 0.5 * (std::atan2(std::abs(radiusY) * std::sin(tgtAngle), std::abs(radiusX) * std::cos(tgtAngle)) + tgtAngle);
```